FHIRClientDevelopmentGuide.md

```diff
@@ -0,0 +1,100 @@
+
+### Configurable Parameters
+- STEVE Username
+- STEVE Password
+- STEVE Client Secret
+- STEVE Client Token
+- STEVE Authentication endpoint
+- SAMS Username
+- SAMS Password
+- SAMS Client Secret
+- SAMS Client Token
+- SAMS Authentication endpoint
+
+## Timeliness
+The client must submit FHIR messages in a timely manner. It is recommended that once a Death Record is entered into the EDRS system, it be submitted as soon as it becomes available, ideally within 1 hour.
+
+The client must support polling the FHIR API to regularly check for FHIR message responses from NCHS. It is recommended to poll the FHIR API at least once every 5 minutes.
+
+### Configurable Parameters
+- Submission interval
+- Polling interval
+
+## Message Resends
+The client must support resends if a message is not acknowledged within the configured time frame. The resubmission message header must be identical to the initial submission message header. The header date timestamp must be identical to the initial message to preserve the order the submission message and any update messages should be processed by NCHS. It is recommended to save the json bundle to preserve the exact message in case there is a need to resubmit. The current recommended resend window is 5 hours.
+
+The maximum number of retries should be 3. After 3 attempts and no acknowledgements, a jurisdictional contact at NCHS should be notified via email or some other notification system. Jurisdictions should identify someone on their end of the system who is responsible for monitoring failed messages who will receive and respond to these notifications.
+
+The current recommendation is to implement an “exponential” back off when resending messages. 
+1.	5 hours
+2.	10 hours
+3.	20 hours
+
+### Configurable Parameters
+- Resend interval
+- Maximum retries
+
+## Response Message Handling
+The client must handle the following message response types
+- Acknowledgements (ACK)
+- Extraction Errors (ERR)
+- Cause of Death Coding Responses (TRX)
+- Demographic Coding Responses (MRE)
+- Status Response (STM)
+
+The client must acknowledge the following message response types
+- Cause of Death Coding Responses (TRX)
+- Demographic Coding Responses (MRE)
+
+The client should support acknowledging response messages in a timely manner to avoid unnecessary duplicate messages from NCHS. Acknowledgements should be sent as soon as they become available and no more than 3 hours after.
+
+### Configurable Parameters
+- Submission interval
+
+## Message Traceabality
+The client must maintain a persistent store of Death Records submitted to NCHS. It is recommended to make the message ID of past records easily accessible for traceability.
+
+The client must support tracing message responses to their initial submission so they can track the status of that message. Use the message ID of the submitted message and the reference message ID in message responses to implement traceability. 
```

Comment:
Maybe FHIR snippets here to show what parts of the messages map? Basically where to find the ID in the submission message and in the ack message?